### PR TITLE
Add release notes for 2.0.0.0-rc1

### PR DESCRIPTION
**Pull request by qreshi**
Signed-off-by: Mohammad Qureshi <47198598+qreshi@users.noreply.github.com>

### Description
Add release notes for 2.0.0.0-rc1

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## Version 2.0.0.0-rc1 2022-04-25
+
+Compatible with OpenSearch 2.0.0-rc1
+
+### Features
+This is the first official release of OpenSearch Notifications.
+
+Notifications consists of `notifications-core` and `notifications` backend plugins for OpenSearch
```
Comment (Naarcha-AWS) [inline release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md:8]: Do we have download links to each?

File: release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+## Version 2.0.0.0-rc1 2022-04-25
+
+Compatible with OpenSearch 2.0.0-rc1
+
+### Features
+This is the first official release of OpenSearch Notifications.
+
+Notifications consists of `notifications-core` and `notifications` backend plugins for OpenSearch 
+and a `notificationsDashboards` frontend plugin for OpenSearch Dashboards.
```
Comment (Naarcha-AWS) [inline release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md:9]: Why is `notificationsDashboards` camelCase vs `notifications-code`?

File: release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+## Version 2.0.0.0-rc1 2022-04-25
+
+Compatible with OpenSearch 2.0.0-rc1
+
+### Features
+This is the first official release of OpenSearch Notifications.
```
Comment (Naarcha-AWS) [inline release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md:6]: Change to "OpenSearch 2.0.0 is the first official release OpenSearch Notifications."

File: release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md (hunk 1 above)
Comment (Naarcha-AWS) [inline release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md:8]: Change to, "Notifications consist of three plugins, `notifications-core` and `notifications` backend plugins for OpenSearch, and a `notificationsDashboards` frontend plugin for OpenSearch Dashboards."

File: release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+Notifications consists of `notifications-core` and `notifications` backend plugins for OpenSearch 
+and a `notificationsDashboards` frontend plugin for OpenSearch Dashboards.
+
+With Notifications you will be able to manage your notification channel configurations in a centralized location
```
Comment (Naarcha-AWS) [inline release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md:11]: Change to, "Use Notifications to manage your notification channel configurations in a centralized location..."

File: release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+and send messages to these channels.
+
+Adds backend REST APIs for CRUD operations on channels and internal transport
```
Comment (Naarcha-AWS) [inline release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md:14]: "This release also adds notification-specifc REST APIs for CRUD operations on channels and internal transport APIs for integrating/communicating with other plugins."

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]
Some changes and a question.

File: release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md (hunk 1 above)
Comment (qreshi) [reply release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md:8]: The documentation hasn't been written yet so I didn't have anything to link to here. It seems from the [opensearch downloads page](https://opensearch.org/downloads.html#opensearch) we only ever link whole distributions or min, not specific artifacts.

File: release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md (hunk 2 above)
Comment (qreshi) [reply release-notes/opensearch-notifications.release-notes-2.0.0.0-rc1.md:9]: It's the convention that was chose it seems for dashboard plugin naming vs backend. I was just using the plugin names so kept it consistent to that.

**Review by Naarcha-AWS — APPROVED** [review]

**Review by lezzago — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Review by dbbaughe — APPROVED** [review]

**Review by rishabhmaurya — APPROVED** [review]

**Pull request merged**